packages/fxa-auth-server/lib/routes/sign.js

```diff
@@ -54,8 +54,9 @@ module.exports = (log, signer, db, domain, devices) => {
         const sessionToken = request.auth.credentials;
         const publicKey = request.payload.publicKey;
         const duration = request.payload.duration;
-        // This is a legacy endpoint that's only used by clients connected to sync,
-        // so always set `service=sync` for metrics logging purposes.
+        // This is a legacy endpoint that's typically only used by
+        // clients connected to sync, so assume `service=sync` for
+        // metrics logging purposes unless we're told otherwise.
         if (!request.query.service) {
```

Comment:
I think you might be able to do `default('sync')` in joi and remove the if block here.

Reply:
Granted it is easy to see what is happening with the comment and the if block so your call.

Reply:
Thanks, yeah, I mostly just left it as-is to keep the change as small as possible.